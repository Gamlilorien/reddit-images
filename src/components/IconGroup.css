.blue-icon {
    filter: invert(54%) sepia(92%) saturate(1963%) hue-rotate(140deg) brightness(103%) contrast(100%);
    height: 25px;
  }
  .blue-icon:hover {
    filter: invert(91%) sepia(88%) saturate(6573%) hue-rotate(161deg) brightness(106%) contrast(105%);
  }