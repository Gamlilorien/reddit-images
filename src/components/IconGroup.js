import React, { useState } from 'react';
import './IconGroup.css';
import imageIcon from '../icons/image-solid.svg';
import linkIcon from '../icons/reddit-square-brands.svg';
import favIcon from '../icons/bookmark-regular.svg';
import favSelectedIcon from '../icons/bookmark-solid.svg';



const IconGroup = (props) => {
    const [isFav, updateFav] = useState(props.entry.isFavorite);
    const entry = props.entry;

    const toggleFavorite = () => {
      //let isFav = entry.isFavorite;
      if(isFav === true) {
        updateFav(false)
      } else {
        updateFav(true)
      }
      updateFavorite(entry.title)
    }

    const updateFavorite = (title) => {
      for(const obj of props.array) {
      if(obj.title === title) {
          let current = obj.isFavorite;
          if(current === true) {
              obj.isFavorite = false;
          } else {
              obj.isFavorite = true;
          } 
        }
      }
    //console.log(props.array)
    }

    return (
        <div className="row">
        <div className="col">
          {
            isFav ? (
              <img src={favSelectedIcon} alt="favorited" className="blue-icon" onClick={toggleFavorite} />
            ) : (
              <img src={favIcon} alt="click to favorite" className="blue-icon" onClick={toggleFavorite} />
            )
          }
          <span> | </span>
          <a href={entry.link} target="_blank" rel="noreferrer">
            <img src={linkIcon} alt={entry.title} className="blue-icon"  />
          </a>
          <span> | </span>
          <a href={entry.image} target="_blank" rel="noreferrer">
            <img src={imageIcon} alt="open full size" className="blue-icon"  />
          </a>
        </div>
      </div>
    )
}

export default IconGroup;