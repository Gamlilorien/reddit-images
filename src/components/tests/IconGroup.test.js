import React from 'react';
import renderer from 'react-test-renderer';
import * as ReactDomClient from 'react-dom/client'
import IconGroup from '../IconGroup';

//sample entry format
let entry = {
    title: "Here Ya Go",
    thumb: "https://b.thumbs.redditmedia.com/i32pGUEIzNKdU6EVifHXopFsIW6jEkZ-aTXdyp4-KkY.jpg",
    image: "https://i.redd.it/ecy7638h8n391.jpg",
    link: "https://www.reddit.com/r/Funnypics/comments/v4u7s7/here_ya_go/"
}
const div = document.createElement("div");
const root = ReactDomClient.createRoot(div);

//afterEach(cleanup);

//make sure that the element gets created regarless of props
it("IconGroup element renders", () => {
    root.render(<IconGroup />)
})


it("IconGroup matches test snapshot", () => {
    const tree = renderer.create(<IconGroup entry={entry} array={entry} />)
    expect(tree).toMatchSnapshot();
})