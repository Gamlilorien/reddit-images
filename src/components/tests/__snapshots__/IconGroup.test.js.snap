// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconGroup matches test snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <img
      alt="click to favorite"
      className="blue-icon"
      onClick={[Function]}
      src="bookmark-regular.svg"
    />
    <span>
       | 
    </span>
    <a
      href="https://www.reddit.com/r/Funnypics/comments/v4u7s7/here_ya_go/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Here Ya Go"
        className="blue-icon"
        src="reddit-square-brands.svg"
      />
    </a>
    <span>
       | 
    </span>
    <a
      href="https://i.redd.it/ecy7638h8n391.jpg"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="open full size"
        className="blue-icon"
        src="image-solid.svg"
      />
    </a>
  </div>
</div>
`;
